lib/benchee.ex

```diff
@@ -55,6 +55,14 @@ for {module, moduledoc} <- [{Benchee, elixir_doc}, {:benchee, erlang_doc}] do
 
     defp add_benchmarking_jobs(suite, jobs) do
       Enum.reduce(jobs, suite, fn {key, function}, suite_acc ->
+        if is_function(function) && :erlang.fun_info(function, :module) == {:module, :erl_eval} do
+          IO.puts("""
+          Warning: the benchmark #{key} is using an evaluated function.
+            Evaluated functions perform slower than compiled functions.
+            Consider moving all of the Benchee caller to a function in a module and invoking the `Mod.fun()` instead.
```

Comment:
In the `Benchee.Benchmark` module you'll see that there is a dedicated printer module to cover printing of text messages (for testability but also separation of concerns).

As for the warning itself, I think we'd also need to tell folks that they can put the benchmark into a `benchmark.exs` file and just `mix run benchmark.exs` - that also works to avoid this. Might bloat up the message a bit too much and we might wanna link to some wiki/README instead. Not sure. See what you think :)